### PR TITLE
feat(event): delete, archive events and delete, archive, unschedule editions

### PR DESCRIPTION
**Pull request by PikinerLiliya**
- Added feature of delete, archive, unschedule of Editions
- Added feature of delete, archive of Events
- Additional check for status 204 in response

### CODE REVIEW

File: src/lib/model/Edition.spec.ts [hunk 1]
```diff
@@ -14,6 +14,30 @@ test('get event', async (t) => {
...
 });
 
+test('archive edition', async (t) => {
+  const client = new MockDynamicContent();
+  const edition = await client.editions.get('5b32379e4cedfd01c4504172');
+  const result = await edition.related.archive();
+  t.is(result.id, '5b32379e4cedfd01c4504172');
+});
+
+test('unshcedult edition', async (t) => {
```
Comment (easen-amp) [inline src/lib/model/Edition.spec.ts:24]: unschedule ?

File: src/lib/model/Event.spec.ts [hunk 2]
```diff
@@ -15,6 +15,23 @@ test('list editions', async (t) => {
...
+  const result = await event.related.archive();
+  t.is(result.id, '5b32379e4cedfd01c4504171');
+});
+
+test('delete event', async (t) => {
+  const client = new MockDynamicContent();
+  const event = await client.events.get('5b32379e4cedfd01c4504171');
+  const result = await event.related.delete();
+
+  console.log(result);
```
Comment (easen-amp) [inline src/lib/model/Event.spec.ts:30]: console.log()

**Review by easen-amp — CHANGES_REQUESTED** [review]

File: src/lib/model/Edition.spec.ts (hunk 1 above)
Comment (PikinerLiliya) [reply src/lib/model/Edition.spec.ts:24]: Changed typo

File: src/lib/model/Event.spec.ts (hunk 2 above)
Comment (PikinerLiliya) [reply src/lib/model/Event.spec.ts:30]: Removed log

File: src/lib/DynamicContent.ts [hunk 3]
```diff
@@ -233,7 +233,7 @@ export class DynamicContent {
   /**
    * @hidden
    */
-  private client: HalClient;
+  public client: HalClient;
```
Comment (rs-amp) [inline src/lib/DynamicContent.ts:236]: Why has this been made public? I can't see it being used anywhere.

File: src/lib/hal/models/HalResource.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export interface HalResourceConstructor<T extends HalResource> {
  */
 export class HalResource {
   // eslint-disable-next-line
-  public _links?: Map<string, HalLink>;
+  public _links?: Record<string, HalLink>;
```
Comment (rs-amp) [inline src/lib/hal/models/HalResource.ts:16]: This is quite a large change, why was it made?

**Review by rs-amp — COMMENTED** [review]
The additional functions look good, but I have a few questions around other changes.

File: src/lib/DynamicContent.ts (hunk 3 above)
Comment (PikinerLiliya) [reply src/lib/DynamicContent.ts:236]: Good point, it's the old logic change that should be reverted. Thanks

File: src/lib/hal/models/HalResource.ts (hunk 4 above)
Comment (PikinerLiliya) [reply src/lib/hal/models/HalResource.ts:16]: Same as above, was used for an older logic, to created a typed map. Will revert it back. Thanks

**Review by rs-amp — APPROVED** [review]
Nice one, LGTM.

**Review by easen-amp — APPROVED** [review]

**Review by Benj0s — APPROVED** [review]

**Review by simonmudd — APPROVED** [review]

**Pull request merged**